{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"100%\"\n",
       "            height=\"650\"\n",
       "            src=\"http://127.0.0.1:8050/\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "            \n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x1c703f8f2d0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=None,\n",
      "    dependents=0,\n",
      "    education=None,\n",
      "    self_employed=None,\n",
      "    applicant_income=0,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender Married  Dependents Education Self_Employed  ApplicantIncome  \\\n",
      "0   None    None           0      None          None                0   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array(['Male'], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array(['Male'], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array(['Male'], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array(['Male'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = 'Male'\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key='Male'\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = 'Male'\n",
      "\n",
      "KeyError: 'Male'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender='Male',\n",
      "    married='No',\n",
      "    dependents=0,\n",
      "    education='Graduate',\n",
      "    self_employed='No',\n",
      "    applicant_income=0,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area='Urban'\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender Married  Dependents Education Self_Employed  ApplicantIncome  \\\n",
      "0   Male      No           0  Graduate            No                0   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0         Urban  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    Male\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    Male\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array(['Male'], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array(['Male'], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array(['Male'], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: 'Male'\n",
      "\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=0,\n",
      "    education=None,\n",
      "    self_employed=None,\n",
      "    applicant_income=0,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents Education Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           0      None          None                0   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=None,\n",
      "    self_employed=None,\n",
      "    applicant_income=0,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents Education Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1      None          None                0   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=None,\n",
      "    applicant_income=0,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0          None                0   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=0,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1                0   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=None,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed ApplicantIncome  \\\n",
      "0   None        1           1          0              1            None   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=1,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1                1   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=10,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1               10   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=100,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1              100   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=1000,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1             1000   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=10000,\n",
      "    coapplicant_income=0,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1            10000   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  0           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=10000,\n",
      "    coapplicant_income=None,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1            10000   \n",
      "\n",
      "  CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0              None           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=10000,\n",
      "    coapplicant_income=1,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1            10000   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                  1           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=10000,\n",
      "    coapplicant_income=12,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1            10000   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                 12           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=10000,\n",
      "    coapplicant_income=122,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1            10000   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0                122           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=10000,\n",
      "    coapplicant_income=1222,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=1,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1            10000   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0               1222           0               360               1   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=10000,\n",
      "    coapplicant_income=1222,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=0,\n",
      "    property_area=None\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1            10000   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0               1222           0               360               0   \n",
      "\n",
      "  Property_Area  \n",
      "0          None  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n",
      "---------------------------------------------------------------------------\n",
      "KeyError                                  Traceback (most recent call last)\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:224, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    223 try:\n",
      "--> 224     return _map_to_integer(values, uniques)\n",
      "        uniques = array(['Rural', 'Urban'], dtype=object)\n",
      "        values = array([None], dtype=object)\n",
      "    225 except KeyError as e:\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in _map_to_integer(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object)\n",
      ")\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        values = array([None], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:164, in <listcomp>(.0=<iterator object>)\n",
      "    163 table = _nandict({val: i for i, val in enumerate(uniques)})\n",
      "--> 164 return np.array([table[v] for v in values])\n",
      "        table = {'Rural': 0, 'Urban': 1}\n",
      "        np = <module 'numpy' from 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\numpy\\\\__init__.py'>\n",
      "        v = None\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:158, in _nandict.__missing__(\n",
      "    self={'Rural': 0, 'Urban': 1},\n",
      "    key=None\n",
      ")\n",
      "    157     return self.nan_value\n",
      "--> 158 raise KeyError(key)\n",
      "        key = None\n",
      "\n",
      "KeyError: None\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "ValueError                                Traceback (most recent call last)\n",
      "Cell In[1], line 154, in update_output(\n",
      "    gender=None,\n",
      "    married=1,\n",
      "    dependents=1,\n",
      "    education=0,\n",
      "    self_employed=1,\n",
      "    applicant_income=10000,\n",
      "    coapplicant_income=1222,\n",
      "    loan_amount=0,\n",
      "    loan_amount_term=360,\n",
      "    credit_history=0,\n",
      "    property_area=0\n",
      ")\n",
      "    126 @app.callback(\n",
      "    127     Output('prediction-output', 'children'),\n",
      "    128     [Input('gender-dropdown', 'value'),\n",
      "   (...)\n",
      "    139 def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
      "    140     # Prepare input data for prediction\n",
      "    141     input_data = pd.DataFrame({\n",
      "    142         'Gender': [gender],\n",
      "    143         'Married': [married],\n",
      "   (...)\n",
      "    152         'Property_Area': [property_area]\n",
      "    153     })\n",
      "--> 154     input_data['Gender'] = le.transform(input_data['Gender'])\n",
      "        input_data =   Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \\\n",
      "0   None        1           1          0              1            10000   \n",
      "\n",
      "   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \\\n",
      "0               1222           0               360               0   \n",
      "\n",
      "   Property_Area  \n",
      "0              0  \n",
      "        le = LabelEncoder()\n",
      "    155     input_data['Married'] = le.transform(input_data['Married'])\n",
      "    156     input_data['Education'] = le.transform(input_data['Education'])\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_set_output.py:140, in _wrap_method_output.<locals>.wrapped(\n",
      "    self=LabelEncoder(),\n",
      "    X=0    None\n",
      "Name: Gender, dtype: object,\n",
      "    *args=(),\n",
      "    **kwargs={}\n",
      ")\n",
      "    138 @wraps(f)\n",
      "    139 def wrapped(self, X, *args, **kwargs):\n",
      "--> 140     data_to_wrap = f(self, X, *args, **kwargs)\n",
      "        f = <function LabelEncoder.transform at 0x000001C703CC31A0>\n",
      "        self = LabelEncoder()\n",
      "        X = 0    None\n",
      "Name: Gender, dtype: object\n",
      "        args = ()\n",
      "        kwargs = {}\n",
      "    141     if isinstance(data_to_wrap, tuple):\n",
      "    142         # only wrap the first output for cross decomposition\n",
      "    143         return (\n",
      "    144             _wrap_data_with_container(method, data_to_wrap[0], X, self),\n",
      "    145             *data_to_wrap[1:],\n",
      "    146         )\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\preprocessing\\_label.py:139, in LabelEncoder.transform(\n",
      "    self=LabelEncoder(),\n",
      "    y=array([None], dtype=object)\n",
      ")\n",
      "    136 if _num_samples(y) == 0:\n",
      "    137     return np.array([])\n",
      "--> 139 return _encode(y, uniques=self.classes_)\n",
      "        y = array([None], dtype=object)\n",
      "        self = LabelEncoder()\n",
      "        self.classes_ = array(['Rural', 'Urban'], dtype=object)\n",
      "\n",
      "File c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\sklearn\\utils\\_encode.py:226, in _encode(\n",
      "    values=array([None], dtype=object),\n",
      "    uniques=array(['Rural', 'Urban'], dtype=object),\n",
      "    check_unknown=True\n",
      ")\n",
      "    224         return _map_to_integer(values, uniques)\n",
      "    225     except KeyError as e:\n",
      "--> 226         raise ValueError(f\"y contains previously unseen labels: {str(e)}\")\n",
      "    227 else:\n",
      "    228     if check_unknown:\n",
      "\n",
      "ValueError: y contains previously unseen labels: None\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-5 (<lambda>):\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\arlow\\anaconda3\\Lib\\threading.py\", line 1045, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\arlow\\AppData\\Roaming\\Python\\Python311\\site-packages\\ipykernel\\ipkernel.py\", line 766, in run_closure\n",
      "    _threading_Thread_run(self)\n",
      "  File \"c:\\Users\\arlow\\anaconda3\\Lib\\threading.py\", line 982, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\dash\\dash.py\", line 1858, in <lambda>\n",
      "    target=lambda: _watch.watch(\n",
      "                   ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\dash\\_watch.py\", line 35, in watch\n",
      "    walk()\n",
      "  File \"c:\\Users\\arlow\\anaconda3\\Lib\\site-packages\\dash\\_watch.py\", line 20, in walk\n",
      "    info = os.stat(path)\n",
      "           ^^^^^^^^^^^^^\n",
      "FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\\\Users\\\\arlow\\\\anaconda3\\\\Lib\\\\site-packages\\\\dash\\\\html\\\\__pycache__\\\\Div.cpython-311-pytest-7.4.0.pyc.4380'\n"
     ]
    }
   ],
   "source": [
    "# Import necessary libraries\n",
    "import dash\n",
    "from dash import dcc, html\n",
    "from dash.dependencies import Input, Output\n",
    "import pandas as pd\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "\n",
    "# Load the dataset\n",
    "data = {\n",
    "    'Loan_ID': ['LP001002', 'LP001003', 'LP001005', 'LP001006', 'LP001008'],\n",
    "    'Gender': ['Male', 'Male', 'Male', 'Male', 'Male'],\n",
    "    'Married': ['No', 'Yes', 'Yes', 'Yes', 'No'],\n",
    "    'Dependents': [0, 1, 0, 0, 0],\n",
    "    'Education': ['Graduate', 'Graduate', 'Graduate', 'Not Graduate', 'Graduate'],\n",
    "    'Self_Employed': ['No', 'No', 'Yes', 'No', 'No'],\n",
    "    'ApplicantIncome': [5849, 4583, 3000, 2583, 6000],\n",
    "    'CoapplicantIncome': [0.0, 1508.0, 0.0, 2358.0, 0.0],\n",
    "    'LoanAmount': [128.0, 128.0, 66.0, 120.0, 141.0],\n",
    "    'Loan_Amount_Term': [360.0, 360.0, 360.0, 360.0, 360.0],\n",
    "    'Credit_History': [1.0, 1.0, 1.0, 1.0, 1.0],\n",
    "    'Property_Area': ['Urban', 'Rural', 'Urban', 'Urban', 'Urban'],\n",
    "    'Loan_Status': ['Y', 'N', 'Y', 'Y', 'Y']\n",
    "}\n",
    "\n",
    "df = pd.DataFrame(data)\n",
    "\n",
    "# Preprocess the data\n",
    "le = LabelEncoder()\n",
    "df['Gender'] = le.fit_transform(df['Gender'])\n",
    "df['Married'] = le.fit_transform(df['Married'])\n",
    "df['Education'] = le.fit_transform(df['Education'])\n",
    "df['Self_Employed'] = le.fit_transform(df['Self_Employed'])\n",
    "df['Property_Area'] = le.fit_transform(df['Property_Area'])\n",
    "\n",
    "# Define features and target\n",
    "X = df.drop(['Loan_ID', 'Loan_Status'], axis=1)\n",
    "y = df['Loan_Status']\n",
    "\n",
    "# Split data into train and test sets\n",
    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)\n",
    "\n",
    "# Train the model\n",
    "model = RandomForestClassifier()\n",
    "model.fit(X_train, y_train)\n",
    "\n",
    "# Initialize the Dash app\n",
    "app = dash.Dash(__name__)\n",
    "\n",
    "# Define the layout\n",
    "app.layout = html.Div([\n",
    "    html.H1(\"Loan Eligibility Prediction\"),\n",
    "    html.Div([\n",
    "        html.Label(\"Gender\"),\n",
    "        dcc.Dropdown(\n",
    "            id='gender-dropdown',\n",
    "            options=[{'label': i, 'value': i} for i in df['Gender']],\n",
    "            value='Male'\n",
    "        ),\n",
    "        html.Label(\"Married\"),\n",
    "        dcc.Dropdown(\n",
    "            id='married-dropdown',\n",
    "            options=[{'label': i, 'value': i} for i in df['Married']],\n",
    "            value='No'\n",
    "        ),\n",
    "        html.Label(\"Dependents\"),\n",
    "        dcc.Dropdown(\n",
    "            id='dependents-dropdown',\n",
    "            options=[{'label': i, 'value': i} for i in df['Dependents']],\n",
    "            value=0\n",
    "        ),\n",
    "        html.Label(\"Education\"),\n",
    "        dcc.Dropdown(\n",
    "            id='education-dropdown',\n",
    "            options=[{'label': i, 'value': i} for i in df['Education']],\n",
    "            value='Graduate'\n",
    "        ),\n",
    "        html.Label(\"Self Employed\"),\n",
    "        dcc.Dropdown(\n",
    "            id='self-employed-dropdown',\n",
    "            options=[{'label': i, 'value': i} for i in df['Self_Employed']],\n",
    "            value='No'\n",
    "        ),\n",
    "        html.Label(\"Applicant Income\"),\n",
    "        dcc.Input(\n",
    "            id='applicant-income-input',\n",
    "            type='number',\n",
    "            value=0\n",
    "        ),\n",
    "        html.Label(\"Coapplicant Income\"),\n",
    "        dcc.Input(\n",
    "            id='coapplicant-income-input',\n",
    "            type='number',\n",
    "            value=0\n",
    "        ),\n",
    "        html.Label(\"Loan Amount\"),\n",
    "        dcc.Input(\n",
    "            id='loan-amount-input',\n",
    "            type='number',\n",
    "            value=0\n",
    "        ),\n",
    "        html.Label(\"Loan Amount Term\"),\n",
    "        dcc.Input(\n",
    "            id='loan-amount-term-input',\n",
    "            type='number',\n",
    "            value=360\n",
    "        ),\n",
    "        html.Label(\"Credit History\"),\n",
    "        dcc.Input(\n",
    "            id='credit-history-input',\n",
    "            type='number',\n",
    "            value=1\n",
    "        ),\n",
    "        html.Label(\"Property Area\"),\n",
    "        dcc.Dropdown(\n",
    "            id='property-area-dropdown',\n",
    "            options=[{'label': i, 'value': i} for i in df['Property_Area']],\n",
    "            value='Urban'\n",
    "        ),\n",
    "    ]),\n",
    "    html.Div(id='prediction-output')\n",
    "])\n",
    "\n",
    "# Define callback to update output\n",
    "@app.callback(\n",
    "    Output('prediction-output', 'children'),\n",
    "    [Input('gender-dropdown', 'value'),\n",
    "     Input('married-dropdown', 'value'),\n",
    "     Input('dependents-dropdown', 'value'),\n",
    "     Input('education-dropdown', 'value'),\n",
    "     Input('self-employed-dropdown', 'value'),\n",
    "     Input('applicant-income-input', 'value'),\n",
    "     Input('coapplicant-income-input', 'value'),\n",
    "     Input('loan-amount-input', 'value'),\n",
    "     Input('loan-amount-term-input', 'value'),\n",
    "     Input('credit-history-input', 'value'),\n",
    "     Input('property-area-dropdown', 'value')])\n",
    "def update_output(gender, married, dependents, education, self_employed, applicant_income, coapplicant_income, loan_amount, loan_amount_term, credit_history, property_area):\n",
    "    # Prepare input data for prediction\n",
    "    input_data = pd.DataFrame({\n",
    "        'Gender': [gender],\n",
    "        'Married': [married],\n",
    "        'Dependents': [dependents],\n",
    "        'Education': [education],\n",
    "        'Self_Employed': [self_employed],\n",
    "        'ApplicantIncome': [applicant_income],\n",
    "        'CoapplicantIncome': [coapplicant_income],\n",
    "        'LoanAmount': [loan_amount],\n",
    "        'Loan_Amount_Term': [loan_amount_term],\n",
    "        'Credit_History': [credit_history],\n",
    "        'Property_Area': [property_area]\n",
    "    })\n",
    "    input_data['Gender'] = le.transform(input_data['Gender'])\n",
    "    input_data['Married'] = le.transform(input_data['Married'])\n",
    "    input_data['Education'] = le.transform(input_data['Education'])\n",
    "    input_data['Self_Employed'] = le.transform(input_data['Self_Employed'])\n",
    "    input_data['Property_Area'] = le.transform(input_data['Property_Area'])\n",
    "    \n",
    "    # Make prediction\n",
    "    prediction = model.predict(input_data)\n",
    "    \n",
    "    return f\"Loan Eligibility Prediction: {prediction[0]}\"\n",
    "\n",
    "# Run the app\n",
    "if __name__ == '__main__':\n",
    "    app.run_server(debug=True)\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
